src/mpi/datatype/type_flatten.c

```diff
@@ -42,12 +41,9 @@ int MPIR_Type_flatten(MPI_Datatype type,
     MPIR_Assert(datatype_ptr->is_committed);
     MPIR_Assert(*array_len_p >= datatype_ptr->max_contig_blocks);
 
-    segp = MPIR_Segment_alloc();
-    err = MPIR_Segment_init(0, 1, type, segp);  /* first 0 is bufptr,
+    segp = MPIR_Segment_alloc(0, 1, type);      /* first 0 is bufptr,
                                                  * 1 is count
                                                  */
-    if (err)
-        return err;
 
```

Comment:
We might want to check `segp != NULL` here. I think this fix can be in a different commit.

Reply:
Done.  Btw, this file will be completely rewritten in a follow-up commit that I have coming.